### PR TITLE
Add support for search files

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Add support for search files

#### Ticket Link
[MM-28019](https://mattermost.atlassian.net/browse/MM-28019)

### CODE REVIEW

**Comment by jespino** [comment]
From QA perspective, this shouldn't affect anything until is used for the new search files feature. Maybe a quick smoke testing around searches is more than enough.

File: src/actions/search.ts [hunk 1]
```diff
@@ -102,7 +127,68 @@ export function getMorePostsForSearch(): ActionFunc {
...
+        dispatch({
+            type: SearchTypes.SEARCH_FILES_REQUEST,
+            isGettingMore,
+        });
+
+        let files: FileSearchResults;
+        try {
+            files = await Client4.searchFilesWithParams(teamId, params);
+
+            // await dispatch(getMissingChannelsFromFiles(files.file_infos));
```
Comment (mgdelacroix) [inline src/actions/search.ts:148]: :eyes:

Comment (jespino) [reply src/actions/search.ts:148]: image

Comment (jespino) [reply src/actions/search.ts]: Done

**Review by mgdelacroix — APPROVED** [review]

**Comment by jespino** [comment]
@ogi-m this is not changing anything important in the rest of the code, is only adding the machinery to access the new API endpoint. I don't think this need testing by itself, it can be tested after merged as part of the whole feature.

**Review by ogi-m — APPROVED** [review]
Thanks @jespino 👍🏻

**Pull request merged**